### ISSUE TITLE
When sending, have a `max` button to send the whole balance (sweep wallet)

### ISSUE THREAD

**Issue by pythcoiner** [wizardsardine/liana#546]
In the send screen, it would be useful to have a `MAX` button that autofill the sum amount of all the wallet coins.
![image](https://github.com/wizardsardine/liana/assets/124568858/afa45ffa-7352-4a39-bf03-5298248ac8c6)

**Comment by darosior** [comment]
Yes. It's something we've discussed. It's a bit intricate to make work as you change the feerate though.

**Comment by pythcoiner** [comment]
> Yes. It's something we've discussed. It's a bit intricate to make work as you change the feerate though.

yes, i think if `max` has been selected (and until send value not changed by the user) the output amount might be implicitly (total - fee)

**Comment by darosior** [comment]
This was brought up today again. A path forward could be to have the recovery command logic be moved to a "sweep" helper and add another command, `sweep`, that takes a single address and sweeps the whole balance there through the primary path.

The integration in the GUI would be more involved. If sweep was used, it should not let the user add more destinations. Nor select coins actually. @edouardparis wdyt?

**Comment by darosior** [comment]
Adding the daemon label as ideally i think the sweep logic should be in the daemon and tested using the functional tests.

**Comment by edouardparis** [comment]
Recovery panel could be renamed sweep and the primary path added to the list of recovery paths.
Wording can be:

2 paths are accessible next block, please select one
- The primary path
  - Signatures from Ed and Antoine can move 11 coins totalling 1.2334 BTC 
-  Recovery path:
   - Signatures from Ed can move 4 coins totalling 0.6244 BTC

**Comment by darosior** [comment]
That seems reasonable and pretty straightforward.
------- Original Message -------

**Comment by pythcoiner** [comment]
> `sweep`, that takes a single address and sweeps the whole balance there through the primary path.

Do you mean to move in batch _one to one_ or consolidate _many to one_?

**Comment by darosior** [comment]
See also https://github.com/wizardsardine/liana/pull/821.

**Comment by darosior** [comment]
Relevant: https://github.com/wizardsardine/liana/pull/863#discussion_r1423661296

**Comment by darosior** [comment]
I'm not sure. Any opinion?

**Comment by jp1ac4** [comment]
I think we could do both. Having the max button on the Send page is intuitive I think and users may expect to see that option there.

We could have the max button even for multiple recipients: for a given recipient, clicking the max button would return the max available for that recipient after taking into account the other recipients.

**Comment by darosior** [comment]
Sounds good. Then let's do the max button first. We can track changing the recovery screen for a sweep screen into another issue.

I think you have access to the beta of Liana Lite (/Online)? There is a max amount there if you want to play with it before starting with the implementation.

**Comment by jp1ac4** [comment]
> We can track changing the recovery screen for a sweep screen into another issue.

This is now in issue https://github.com/wizardsardine/liana/issues/994.
